### ISSUE TITLE
Test failures with Java 11

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-extras#424]
```
org.eclipse.xtext.common.types.access.reflect.ReflectionTypeProviderTest > testBug337307 FAILED
    java.lang.AssertionError at ReflectionTypeProviderTest.java:210

org.eclipse.xtext.common.types.access.reflect.ReflectionTypeProviderTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.reflect.CachingReflectionTypeProviderTest > testBug337307 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.reflect.CachingReflectionTypeProviderTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.reflect.JarReflectionTypeProviderTest > testBug337307 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.reflect.JarReflectionTypeProviderTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.impl.ClasspathTypeProviderTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.impl.ClasspathTypeProviderWithoutASMTest > testBug337307 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.impl.ClasspathTypeProviderWithoutASMTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

org.eclipse.xtext.common.types.access.impl.JarClasspathTypeProviderTest > testFindTypeByName_javaLangCharSequence_02 FAILED
    java.lang.AssertionError

2175 tests completed, 10 failed
```

```
java.lang.AssertionError: expected:<6> but was:<7>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.eclipse.xtext.common.types.access.impl.AbstractTypeProviderTest.assertMembers(AbstractTypeProviderTest.java:229)
	at org.eclipse.xtext.common.types.access.impl.AbstractTypeProviderTest.testFindTypeByName_javaLangCharSequence_02(AbstractTypeProviderTest.java:322)
```

**Comment by cdietrich** [comment]
its also ugly that we dont print the actual in the assertion message

**Comment by szarnekow** [comment]
There is a new method on the CharSequence interface. The tests should probably be adjusted to rely only on locally defined types.

**Comment by cdietrich** [comment]
or at least non static members

**Comment by szarnekow** [comment]
This particular bug can be closed now, what do you think?

**Comment by cdietrich** [comment]
there are also longrunning tests failing

```

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchClosureTypeTest > testClosureWithReturnExpression_01 FAILED
    org.junit.ComparisonFailure
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchClosureTypeTest > testClosureWithReturnExpression_02 FAILED
    org.junit.ComparisonFailure
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchClosureTypeTest > testClosureWithReturnExpression_03 FAILED
    org.junit.ComparisonFailure
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchClosureTypeTest > testClosureWithReturnExpression_04 FAILED
    org.junit.ComparisonFailure
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchFeatureCallTypeTest > testBug_406425_01 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.IsolationTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchFeatureCallTypeTest > testBug_406425_01 FAILED
    org.junit.ComparisonFailure at BatchFeatureCallTypeTest.java:50

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeArgumentTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure at BatchTypeArgumentTest.java:66

org.eclipse.xtext.xbase.tests.typesystem.RecomputingBatchTypeArgumentTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testIfExpression_04 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:62

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.PermutingBatchReturnTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure at PermutingBatchReturnTypeResolverTest.java:39

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingIsolationTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testIfExpression_04 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchReturnTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.RecomputingTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.InvariantCheckingBatchTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testOverloadedOperators_20 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testTypeGuardedCase_0 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testTypeGuardedCase_3 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testIfExpression_02 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testIfExpression_09 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testIfExpression_11 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.ShuffledTypeResolverTest > testIfExpression_13 FAILED
    org.junit.ComparisonFailure

org.eclipse.xtext.xbase.tests.typesystem.BatchClosureTypeTest > testClosureWithReturnExpression_01 FAILED
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:69
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:88

org.eclipse.xtext.xbase.tests.typesystem.BatchClosureTypeTest > testClosureWithReturnExpression_02 FAILED
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:69
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:88

org.eclipse.xtext.xbase.tests.typesystem.BatchClosureTypeTest > testClosureWithReturnExpression_03 FAILED
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:69
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:88

org.eclipse.xtext.xbase.tests.typesystem.BatchClosureTypeTest > testClosureWithReturnExpression_04 FAILED
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:69
    org.junit.ComparisonFailure at BatchClosureTypeTest.java:88

```

**Comment by cdietrich** [comment]
Fixed in 2.18
